Backend/AlgorithmsManager.py

```diff
@@ -249,6 +275,48 @@ def get_trend_news(self, trend, num_stories=3):
         #todo support fetching multiple pages of results for over 100 stories?
         return example_news_stories
 
+    # TODO: this function adds a TrendSnapshot object to the database with proper format
+    def add_trends_snapshot_to_database(self, snap):
+        highid = self.get_highest_id_of_database_table('trends_snapshot', idname='id')
+
+        querystr = "insert into trends_snapshot(id, woe_id, trend_content, query_term, tweet_volume, is_hashtag, created_date) values \n"
+
+        i = 0
+        trendlen = len(snap.trends)
+        for trend in snap.trends:
+            i += 1
+            highid += 1
+
+            querystr += "({}, {}, '{}', '{}', {}, {}, '{}')".format(highid,
+                                                                    snap.woeid,
+                                                                    trend['trend_content'].replace("%", "%%").replace("\'", "\'\'"),
+                                                                    trend['query_term'].replace("%", "%%").replace("\'", "\'\'"),
+                                                                    trend['tweet_volume'] if trend['tweet_volume'] is not None else 0,
+                                                                    trend['is_hashtag'],
+                                                                    snap.timestamp)  # we want to avoid single % symbol in sql queries
+            if i != trendlen:
+                querystr += ',\n'
+            else:
+                querystr += ';'
+
+        print("Adding", trendlen, "trends to snapshot table")
+        #print(querystr)
+        self.database.query(querystr)
+
+    def get_trends_snapshot_from_database(self, trends, fromdate, todate=datetime.now(), woeid=0):
+        querytemplate = "SELECT * FROM trends_snapshot " \
+                        "WHERE created_date >= '{}' AND created_date < '{}' " \
+                        "AND trend_content LIKE '%%{}%%' " \
```

Comment:
This is to make up for the lack of trends data that we have. Should I change it to  = ?

Reply:
I think so, yes. Otherwise we'd be feeding misleading information to the UI whenever there were two trends such that one trend was a substring of the other.